transports/jms/src/test/resources/integration/jms-request-response.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:jms="http://www.mulesoft.org/schema/mule/jms"
+      xmlns:test="http://www.mulesoft.org/schema/mule/test"
+      xsi:schemaLocation="
+       http://www.mulesoft.org/schema/mule/test http://www.mulesoft.org/schema/mule/test/current/mule-test.xsd
+       http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+       http://www.mulesoft.org/schema/mule/jms http://www.mulesoft.org/schema/mule/jms/current/mule-jms.xsd">
+
+    <flow name="JMSNoCorrelationIDTemporaryQueue">
+        <test:assert expression="#[message.inboundProperties.JMSCorrelationID == null]"/>
+        <jms:outbound-endpoint exchange-pattern="request-response" queue="in" connector-ref="jmsConnector1"/>
```

Comment:
Shouldn't you be using a different queue for each scenario, otherwise any target flow could be picking up this message couldn't it?